lib/routes/profile.js

```diff
@@ -43,27 +56,39 @@ module.exports = {
     }
   },
   handler: function email(req, reply) {
+    const server = req.server;
     var creds = req.auth.credentials;
     if (!hasAllowedScope(creds.scope || [])) {
       return reply(Boom.forbidden());
     }
-    batch(req, {
-      email: '/v1/email',
-      uid: '/v1/uid',
-      avatar: '/v1/avatar',
-      displayName: '/v1/display_name'
-    })
-    .done(function (result) {
-      if (creds.scope.indexOf('openid') !== -1) {
-        result.sub = creds.user;
-      }
-      var rep = reply(result);
-      var etag = computeEtag(result);
-      if (etag) {
-        rep = rep.etag(etag);
+
+    if (!server.methods.batch) {
+      createServerMethod(server);
```

Comment:
The `generateKey` in the function uses whatever args the method we're registering uses. i.e `req` in this case. 

If it feels confusing would moving the method to be registered within the if block itself instead of a different function. Would that seem cleaner?